### PR TITLE
Run sdp-service and sdp-dnmasq as non-privileged users

### PR DESCRIPTION
**Pull request by lepiolet**

### CODE REVIEW

**Review by userappgate — APPROVED** [review]
Seems fine

File: k8s/chart/templates/configmap-sidecar.yaml [hunk 1]
```diff
@@ -17,9 +17,8 @@ data:
           "image": "{{ default .Values.global.image.repository .Values.sdp.headlessService.image.repository }}/sdp-headless-service:{{ default (include "sdp-k8s-client.defaultTag" .) .Values.sdp.headlessService.image.tag}}",
           "imagePullPolicy": "{{.Values.sdp.headlessService.image.pullPolicy}}",
           "securityContext": {
-            "runAsGroup": 0,
-            "runAsNonRoot": false,
-            "runAsUser": 0
+            "runAsGroup": 102,
+            "runAsUser":103
```
Comment (thomascellerier) [inline k8s/chart/templates/configmap-sidecar.yaml:21]: is runAsNonRoot true by default?

**Review by thomascellerier — APPROVED** [review]

File: k8s/chart/templates/configmap-sidecar.yaml (hunk 1 above)
Comment (lepiolet) [reply k8s/chart/templates/configmap-sidecar.yaml:21]: that's to allow the container to run as root

Comment (lepiolet) [reply k8s/chart/templates/configmap-sidecar.yaml:21]: then it needs to be explicit set to True i think

**Review by mandopaloooza — APPROVED** [review]
LGTM

**Review by thomascellerier — APPROVED** [review]

File: k8s/chart/templates/rbac.yaml [hunk 2]
```diff
@@ -37,6 +37,40 @@ roleRef:
...
 apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "sdp-k8s-client.name" . }}-services
+  namespace: "kube-system"
+  labels:
+    {{- include "sdp-k8s-client.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "1"
```
Comment (mandopaloooza) [inline k8s/chart/templates/rbac.yaml:48]: Does this need to be preinstall hooks? Same question for RoleBinding, ClusterRole, ClusterRoleBinding, and ConfigMap

Comment (lepiolet) [reply k8s/chart/templates/rbac.yaml:48]: I don't think it's needed, the roles can be even created after the admission hook is deployed.

**Comment by anoncam** [comment]
@lepiolet yeah, this is awesome guys

**Pull request merged**